frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -218,413 +223,520 @@ export function DynamicConfiguration () {
         setFields({ 'configuration.EMRConfig.applications': updatedSelectedApps });
     };
 
-    return (
-        <Container
-            header={
-                <Header
-                    variant='h2'
-                    description={`The current dynamic configuration of ${window.env.APPLICATION_NAME}. These settings can be modified without redeploying the application.`}
-                >
-                    {window.env.APPLICATION_NAME} Dynamic Configuration
-                </Header>
+
+    const isObject = (x) => typeof x === 'object' && !Array.isArray(x) && x !== null;
+
+    /**
+     * Computes the difference between two JSON objects, recursively.
+     *
+     * This function takes two JSON objects as input and returns a new object that
+     * contains the differences between the two. Works with nested objects.
+     *
+     * @param {object} [obj1={}] - The first JSON object to compare.
+     * @param {object} [obj2={}] - The second JSON object to compare.
+     * @returns {object} - A new object containing the differences between the two input objects.
+     */
+    function getJsonDifference (obj1 = {}, obj2 = {}) {
+        const output = {},
+            merged = { ...obj1, ...obj2 }; // has properties of both
+
+        for (const key in merged) {
+            const value1 = obj1[key], value2 = obj2[key];
```

Comment:
If this is a utility it should account for the fact that the key may not exist. This expects both objects to have the same top-level keys

Reply:
it already does handle that. if they key does not exist, it returns undefined which everything works normally if its undefined. `isObject(value1)` would just return `false` and then `JSON.stringify(value1)` would make undefined into a string 'undefined' so it all works normally. When we recursively call `getJsonDifference(value1, value2)` with an undefined, the return is always an object so the `merged = { ...obj1, ...obj2};` part will work fine.

```
console.log(getJsonDifference({test: 'wa', test2: 'na'}, {test: 'wa'}));

>> {test2: undefined}
```
Which is the expected output.